pinery-miso/src/main/java/ca/on/oicr/pinery/lims/miso/MisoClient.java

```diff
@@ -0,0 +1,1031 @@
+       * @return
+       * @throws SQLException
+       */
+      public Attribute extractAttributeFrom(ResultSet rs) throws SQLException {
+        String val = extractStringValueFrom(rs);
+        return val == null ? null : makeAttribute(getAttributeKey(), extractStringValueFrom(rs));
+      };
+
+      /**
+       * Extracts the value belonging to this AttributeKey from a ResultSet
+       * 
+       * @param rs
+       *          ResultSet containing data to populate this field
+       * @return the value to associate with this AttributeKey; null if absent
+       * @throws SQLException
+       */
+      public String extractStringValueFrom(ResultSet rs) throws SQLException {
+        return rs.getString(getSqlKey());
+      }
+
+      private static Attribute makeAttribute(String name, String value) {
+        Attribute att = new DefaultAttribute();
+        att.setName(name);
+        att.setValue(value);
+        return att;
+      }
+
+      private static enum StrStatus {
+        NOT_SUBMITTED("Not Submitted"), SUBMITTED("Submitted"), PASS("Pass"), FAIL("Fail");
+
+        private final String value;
+
+        private StrStatus(String value) {
+          this.value = value;
+        }
+
+        public String getValue() {
+          return value;
+        }
+      }
+
+    }
+
+  }
+
+  private static class RunSampleRowMapper implements RowMapper<MisoRunSample> {
+
+    @Override
+    public MisoRunSample mapRow(ResultSet rs, int rowNum) throws SQLException {
+      MisoRunSample s = new MisoRunSample();
+
+      s.setId(rs.getString("dilutionId"));
+      s.setPartitionId(rs.getInt("partitionId"));
+      s.setBarcode(AttributeKey.BARCODE.extractStringValueFrom(rs));
+      s.setBarcodeTwo(AttributeKey.BARCODE_TWO.extractStringValueFrom(rs));
+
+      Attribute att = AttributeKey.TARGETED_RESEQUENCING.extractAttributeFrom(rs);
+      if (att != null) {
+        Set<Attribute> atts = new HashSet<>();
+        atts.add(att);
+        s.setAttributes(atts);
+      }
+
+      return s;
+    }
+
+  }
+
+  private static class OrderSampleRowMapper implements RowMapper<MisoOrderSample> {
+
+    private static final AttributeKey[] orderSampleAtts = new AttributeKey[] { AttributeKey.READ_LENGTH,
+        AttributeKey.TARGETED_RESEQUENCING };
+
+    @Override
+    public MisoOrderSample mapRow(ResultSet rs, int rowNum) throws SQLException {
+      MisoOrderSample s = new MisoOrderSample();
+
+      s.setId(rs.getString("libraryId"));
+      s.setOrderId(rs.getInt("orderId"));
+      s.setBarcode(AttributeKey.BARCODE.extractStringValueFrom(rs));
+      s.setBarcodeTwo(AttributeKey.BARCODE_TWO.extractStringValueFrom(rs));
+
+      Set<Attribute> atts = new HashSet<>();
+      for (AttributeKey possibleAtt : orderSampleAtts) {
+        Attribute att = possibleAtt.extractAttributeFrom(rs);
+        if (att != null) {
+          atts.add(att);
+        }
+      }
+      if (atts.size() > 0) {
```

Comment:
![MINOR](https://sonarsource.github.io/sonar-github/severity-minor.png 'Severity: MINOR') Use isEmpty() to check whether the collection is empty or not. [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonarcloud.io/coding_rules#rule_key=squid%3AS1155)